### ISSUE TITLE
[bug] prefer-read-only-props doesn't work well with interface extension & library types

### ISSUE THREAD

**Issue by rdjpalmer** [Rel1cx/eslint-react#1326]
### Describe the bug

`prefer-read-only-props` rule does not work well with extended types. Coming from a react-native/expo project we have a lot of the following code:

```tsx
interface ComponentProps extends [NativeComponentProps] {
  someProp: string
}

function Component(props: ComponentProps) {
  const { someProp, ...nativeProps } = props;
}
```

Where `NativeComponentProps` is say `PressableProps` or `ViewProps`. These come with `react-native` and of course aren't readonly so the rule fails.

I added a generic to force them to be read only like so:

```tsx
type DeepReadOnly<T> = {
  readonly [P in keyof T]: T[P] extends (infer U)[]
    ? ReadonlyArray<DeepReadOnly<U>>
    : T[P] extends ReadonlyArray<infer U>
      ? ReadonlyArray<DeepReadOnly<U>>
      : T[P] extends object
        ? DeepReadOnly<T[P]>
        : T[P];
};

type ReadOnlyPressableProps = DeepReadOnly<Pressable>;
```

However, while this appears to work when you inspect the types of `ReadOnlyPressableProps`, the eslint rule still reports the issue.

### Reproduction

1. Clone [eslint-plugin-react-x-prefer-read-only-props-false-negative](https://github.com/rdjpalmer/eslint-plugin-react-x-prefer-read-only-props-false-negative)
2. Run `npm install --force` (required to ignore some peer dependency issues)
3. Open in your editor
4. Open `src/Component.tsx`
5. You should now see an eslint warning that `Component`'s props are not readonly.
Alternatively, using eslint's cli:

1. Clone [eslint-plugin-react-x-prefer-read-only-props-false-negative](https://github.com/rdjpalmer/eslint-plugin-react-x-prefer-read-only-props-false-negative)
2. Run `npm install --force` (required to ignore some peer dependency issues)
3. Run `npm run lint`

### Expected behavior

No warning should be reported

### Platform and versions

```Text
node -v
v22.14.0



{
  "eslint": "^9.39.1",
  "eslint-plugin-react-x": "^2.3.7",
  "typescript-eslint": "^8.47.0"
}
```

### Stack trace

```Text

```

### Additional context

Note that if you use `type` instead of `interface` the issue seems to go away:

```tsx
type DeepReadOnly<T> = {
  readonly [P in keyof T]: T[P] extends (infer U)[]
    ? ReadonlyArray<DeepReadOnly<U>>
    : T[P] extends ReadonlyArray<infer U>
      ? ReadonlyArray<DeepReadOnly<U>>
      : T[P] extends object
        ? DeepReadOnly<T[P]>
        : T[P];
};

interface PressableProps {
  testID: string;
}

type ReadonlyPressableProps = DeepReadOnly<PressableProps>;

type ComponentProps = {
  readonly name: string;
} & ReadonlyPressableProps;
```

**Comment by Rel1cx** [comment]
This is a very interesting bug. I will look into it as soon as I have time.
